src/visualization/parsers/layout.ts

```diff
@@ -161,7 +161,13 @@ export default class Layout implements IParser {
         var node = <HTMLElement>this.layouts[state.index];
         if (node) {
             // First remove all its existing attributes
-            while (node.attributes && node.attributes.length > 0) node.removeAttribute(node.attributes[0].name);
+            if (node.attributes) {
+                var len = node.attributes.length;
+                while (node.attributes && len > 0) {
+                    node.removeAttribute(node.attributes[0].name);
+                    len--;
+                }
+            }
```

Comment:
this is an odd change for this CL, was this a perf optimization?

Reply:
Edge handles these while loops differently.  node.attributes.length doesn't get updated when an node.removeAttribute is called, so the while would loop infinitely.

---
In reply to: [139212298](https://github.com/Microsoft/clarity-tools/pull/11#discussion_r139212298) [](ancestors = 139212298)